sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -133,14 +133,34 @@ class SparkSession private(
   /** Tag to mark all jobs owned by this session. */
   private[sql] lazy val sessionJobTag = s"spark-session-$sessionUUID"
 
+  /**
+   * A UUID that is unique on the thread level. Used by managedJobTags to make sure that a same
+   * tag from two threads does not overlap in the underlying SparkContext/SQLExecution.
+   */
+  private[sql] lazy val threadUuid = new InheritableThreadLocal[String] {
```

Comment:
Could we have a simple test for this inheritance? Otherwise, looks fine from a cursory look.

Reply:
Yes we do! See
https://github.com/apache/spark/pull/48906/files#diff-647cd00e281c3cccb76a42646a692abaef6738c8157daf6642bc8fa16e233394R174-R175